src/nrc/datamodel/admin.py

```diff
@@ -45,11 +49,112 @@ def get_object_actions(self, obj):
+    """
+    Make a request to the callback URLs of all selected Abonnementen and store the results
+    in the session
+
+    Any subsequent executions of this action will remove the previous results from the session
+    """
+    validator = CallbackURLValidator("callback_url", "auth")
+    callback_statuses = {}
+    for obj in queryset.iterator():
+        # Any other Abonnement that has the same callback URL and auth will be skipped
+        # and gets the same status
+        key = str((obj.callback_url, obj.auth))
+        if key in callback_statuses:
+            continue
+
+        try:
+            validator({"callback_url": obj.callback_url, "auth": obj.auth}, None)
+            callback_statuses[key] = True
+        except (ValidationError, RequestException):
+            callback_statuses[key] = False
+
+    messages.add_message(
+        request,
+        messages.SUCCESS,
+        _(
+            "Retrieve status for selected subscriptions. "
+            "All previous results have been reset."
+        ),
+    )
+    request.session["callback_statuses"] = callback_statuses
+
+
+class StatusCodeFilter(admin.SimpleListFilter):
+    title = "callback URL reachable?"
+    parameter_name = "callback_url_reachable"
+
+    def lookups(self, request, model_admin):
+        return [("true", "Yes"), ("false", "No"), ("unknown", "Unknown")]
+
+    def queryset(self, request, queryset):
+        if not self.value():
+            return queryset
+
+        filtered_ids = []
+        if callback_statuses := request.session.get("callback_statuses"):
+            for obj in queryset.iterator():
+                callback_status = callback_statuses.get(
+                    str((obj.callback_url, obj.auth)), None
+                )
+                if self.value() == "true" and callback_status:
+                    filtered_ids.append(obj.id)
+                elif self.value() == "false" and callback_status == False:  # noqa
+                    filtered_ids.append(obj.id)
+                elif self.value() == "unknown" and callback_status is None:
+                    filtered_ids.append(obj.id)
+            return queryset.filter(id__in=filtered_ids)
+        return queryset
+
+
 @admin.register(Abonnement)
 class AbonnementAdmin(admin.ModelAdmin):
-    list_display = ("uuid", "client_id", "callback_url", "get_kanalen_display")
+    list_display = (
+        "uuid",
+        "client_id",
+        "callback_url",
+        "get_callback_url_reachable",
+        "get_kanalen_display",
+    )
     readonly_fields = ("uuid",)
+    list_filter = (StatusCodeFilter,)
     inlines = (FilterGroupInline,)
+    actions = [check_callback_url_status]
+
+    def changelist_view(self, request, extra_context=None):
+        # Store the request object to ensure the custom admin field has access to it
+        self._request = request
+        return super().changelist_view(request, extra_context=extra_context)
+
+    def check_all_callback_urls(self, request):
```

Comment:
Tbh I think having both action and button is a little excessive. When you select objects for the action you can select everything (even if the number of objects is more than 100)
But it's up to you, so treat this comment as optional

Reply:
I added the button to make it a bit more visible and easy for admins, since Alex mentioned that the existence of these kinds of actions is not always clear right away: https://github.com/open-zaak/open-notificaties/issues/108#issuecomment-2391763531.

Reply:
Thanks for the explanation!